src/databricks/labs/ucx/hive_metastore/udfs.py

```diff
@@ -0,0 +1,110 @@
+import logging
+from collections.abc import Iterable, Iterator
+from dataclasses import dataclass
+from functools import partial
+
+from databricks.labs.blueprint.parallel import Threads
+
+from databricks.labs.ucx.framework.crawlers import CrawlerBase, SqlBackend
+from databricks.labs.ucx.mixins.sql import Row
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class Udf:
+    catalog: str
+    database: str
+    name: str
+    func_type: str
+    func_input: str
+    func_returns: str
+    deterministic: bool
+    data_access: str
+    comment: str
+    body: str
+
+    @property
+    def key(self) -> str:
+        return f"{self.catalog}.{self.database}.{self.name}".lower()
+
+
+class UdfsCrawler(CrawlerBase):
+    def __init__(self, backend: SqlBackend, schema):
+        """
+        Initializes a UdfsCrawler instance.
+
+        Args:
+            backend (SqlBackend): The SQL Execution Backend abstraction (either REST API or Spark)
+            schema: The schema name for the inventory persistence.
+        """
+        super().__init__(backend, "hive_metastore", schema, "udfs", Udf)
+
+    def _all_databases(self) -> Iterator[Row]:
+        yield from self._fetch("SHOW DATABASES")
+
+    def snapshot(self) -> list[Udf]:
+        """
+        Takes a snapshot of tables in the specified catalog and database.
+
+        Returns:
+            list[Table]: A list of Udf objects representing the snapshot of tables.
+        """
+        return self._snapshot(partial(self._try_load), partial(self._crawl))
```

Comment:
You don't need partial, as no arguments are passed down to tryload/crawl

Reply:
corrected